hibernate-core/src/main/java/org/hibernate/dialect/SybaseDialect.java

```diff
@@ -49,6 +49,10 @@ public SybaseDialect(int version) {
 		this.version = version;
 		//Sybase ASE didn't introduce bigint until version 15.0
 		registerColumnType( Types.BIGINT, "numeric(19,0)" );
+
+		//Sybase 'bit' type is always exactly one bit i.e. doesn't support null
+		registerColumnType( Types.BIT, "smallint" );
+		registerColumnType( Types.BOOLEAN, "smallint" );
```

Comment:
I *think*, but I'm not 100% sure, that we can use `bit` for `BOOLEAN` on Sybase Anywhere. It looks like it's only ASE where we can't use it.

Reply:
You are right, SQL Anywhere supports marking the BIT type as `NULL`. I'll fix that.